unittests/BuildSystem/BuildSystemTaskTests.cpp

```diff
@@ -37,6 +38,60 @@ using namespace llbuild::unittests;
 
 namespace {
 
+class LoggingFileSystem : public FileSystem {
+private:
+  std::mutex deletedPathsMutex;
+  std::vector<std::string> deletedPaths;
+  std::mutex missingPathsMutex;
+  std::vector<std::string> missingPaths;
+
+  std::unique_ptr<FileSystem> realFS = createLocalFileSystem();
+
+public:
+  std::vector<std::string> getDeletedPaths() {
+    {
```

Comment:
You don't need the extra scope `{` here, since the entire function is a scope already.

Reply:
Ah right, c&p fail - thanks.